src/nvim/os/env.c

```diff
@@ -169,6 +178,64 @@ void os_get_hostname(char *hostname, size_t size)
 #endif
 }
 
+/// @return -1 if error encountered, 1 if capslock active otherwise 0 if not.
+int capslock_status(void ) {
+
+#if defined(__linux__)
+
+  FILE *fp = NULL;
+  char res[1024];
+  fp = popen("xset -q | grep Caps", "r");
+  if (fp == NULL) {
+      return -1;
+  }
+  fgets(res, sizeof(res), fp);
+  if (STRNCMP(res + 21, "on", 2) == 0) {
+    return 1;
+  } else {
+    return 0;
+  }
+
+#elif defined(WIN32)
+  if ((GetKeyState(VK_CAPITAL) & 0x0001)!=0) {
+    return 1
+  } else {
+    return 0;
+  }
+#elif defined(__APPLE__)
+
+  kern_return_t kr;
```

Comment:
Prefer to declare variables where they're used instead of at the top of a function.  We specifically require C99 support, so let's take advantage of it.

Reply:
OK sure